backend/tests/test_auth_real.py

```
@@ -20,15 +31,112 @@ async def test_get_current_user_rejects_missing_auth():
+
+@pytest.mark.asyncio
+async def test_dev_header_trust_rejects_wrong_token():
+    settings.TRUST_DEV_HEADERS = True
+    settings.DEV_AUTH_TOKEN = SecretStr("expected-dev-auth-token")
+
+    with pytest.raises(HTTPException) as exc:
+        await get_auth_context(
+            x_user_id="attacker",
+            x_user_role="platform_admin",
+            x_organization_id="org-victim",
+            x_dev_auth_token="wrong-dev-auth-token",
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "x_dev_auth_token"

(S106)